Production_scripts/B01_acute_draft.R

```diff
@@ -0,0 +1,332 @@
+  convert_eng_gpprac_to_dummy(gpprac) %>%
+  # Calculate the total length of stay (for the entire episode, not just within the financial year).
+  mutate(
+    stay = difftime(record_keydate2, record_keydate1, units = "days"),
+    # create and populate SMRType
+    SMRType = case_when(
+      recid == "01B" & lineno != 330 ~ if_else(ipdc == "I", "Acute-IP", "Acute-DC")
+      lineno == 330 & ipdc == "I" ~ "GLS-IP",
+      recid == "GLS" ~ "GLS-IP"
+    )
+  ) %>%
+  # Apply new costs for C3 specialty, these are taken from the 2017/18 file
+  fix_c3_costs(year) %>%
+  # initialise monthly cost/beddays variables in a separate data frame for matching
+  convert_monthly_rows_to_vars(uri, costmonthnum, cost_total_net, yearstay) %>%
+  # Add oldtadm as a factor with labels
+  mutate(oldtadm = factor(oldtadm,
+    levels = c(0:8),
+    labels = c(
+      "Deferred admission",
+      "Waiting list/diary/booked",
+      "Repeat admission",
+      "Transfer",
+      "Emergency deliberate self inflicted injury or poisoning",
+      "Emergency road traffic accident",
+      "Emergency home accident (includes accidental poisoning in the home)",
+      "Emergency other injury (includes accidental poisoning other than in the home)",
+      "Emergency other (excluding accidental poisoning)"
+    )
+  ))
+
+
+## save outfile ---------------------------------------
+outfile <- final_acute_file %>%
+  select(
+    year,
+    recid,
+    record_keydate1,
+    record_keydate2,
+    SMRType,
+    chi,
+    gender,
+    dob,
+    gpprac,
+    hbpraccode,
+    postcode,
+    hbrescode,
+    lca,
+    HSCP,
+    DataZone,
+    location,
+    hbtreatcode,
+    yearstay,
+    stay,
+    ipdc,
+    spec,
+    sigfac,
+    conc,
+    mpat,
+    cat,
+    tadm,
+    adtf,
+    admloc,
+    oldtadm,
+    disch,
+    dischto,
+    dischloc,
+    diag1,
+    diag2,
+    diag3,
+    diag4,
+    diag5,
+    diag6,
+    op1a,
+    op1b,
+    dateop1,
+    op2a,
+    op2b,
+    dateop2,
+    op3a,
+    op3b,
+    dateop3,
+    op4a,
+    op4b,
+    dateop4,
+    smr01_cis_marker,
+    age,
+    cij_marker,
+    cij_admtype,
+    cij_ipdc,
+    cij_pattype_code,
+    cij_adm_spec,
+    cij_dis_spec,
+    CIJ_start_date,
+    CIJ_end_date,
+    alcohol_adm,
+    submis_adm,
+    falls_adm,
+    selfharm_adm,
+    commhosp,
+    cost_total_net,
+    apr_beddays,
+    may_beddays,
+    jun_beddays,
+    jul_beddays,
+    aug_beddays,
+    sep_beddays,
+    oct_beddays,
+    nov_beddays,
+    dec_beddays,
+    jan_beddays,
+    feb_beddays,
+    mar_beddays,
+    apr_cost,
+    may_cost,
+    jun_cost,
+    jul_cost,
+    aug_cost,
+    sep_cost,
+    oct_cost,
+    nov_cost,
+    dec_cost,
+    jan_cost,
+    feb_cost,
+    mar_cost,
+    uri
+  ) %>%
+  arrange(chi, record_keydate1)
+
+# Save as zsav file
+outfile %>%
+  readr::write_rds(get_source_extract_path(year, "Acute", ext = "zsav"))
```

Comment:
```suggestion
  haven::write_sav(get_source_extract_path(year, "Acute", ext = "zsav"))
```

will this not need to be using `haven::write_sav` rather than `readrr:write_rds` as this is saving a .sav?